java/vector/src/main/java/org/apache/arrow/vector/BaseValueVector.java

```diff
@@ -122,5 +123,49 @@ public static boolean checkBufRefs(final ValueVector vv) {
   public BufferAllocator getAllocator() {
     return allocator;
   }
+
+  protected void compareTypes(BaseValueVector target, String caller) {
+    if (this.getMinorType() != target.getMinorType()) {
+      throw new UnsupportedOperationException(caller + " should have vectors of exact same type");
```

Comment:
was this in the API before, if not maybe it would be better to  just return a boolean and the caller can throw an exception?

Reply:
Since it's a protected method. I don't think it matters too much. I am ok with it as is.